manager.go

```diff
@@ -95,6 +96,35 @@ func (m *Manager) LoadConfig(cfgPath string) error {
 	return nil
 }
 
+var ErrMissingCloudCreds = errors.New("can't create NetAppender without cloud creds")
+
+// create and attach a NetAppender.
+func (m *Manager) attachNetAppender() error {
+	if m.conn == nil {
+		return errors.New("NetAppender requires non-null conn")
+	}
+	if m.netAppender != nil {
+		return errors.New("Manager already has non-nil netAppender")
+	}
+	if m.cloudAddr == "" || m.partID == "" || m.cloudSecret == "" {
+		return ErrMissingCloudCreds
+	}
```

Comment:
Does this need to be a separate function? When I suggested adding this to dial, I meant to simplify things. At the point this is called (bottom of dial()), all these error checks are impossible to hit, as dial() would've already returned an error.